.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,191 @@
+name: GH_CI
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  main_tests:
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    runs-on: ${{ matrix.os }}
+    strategy:
+        matrix:
+          os: [ubuntu-latest, ]
+          python-version: [3.6, 3.7, 3.8]
+          run_type: [full, ]
+          install_hole: [True, ]
+          codecov: [True, ]
+          include:
+            - name: macOS
+              os: macOS-latest
+              python-version: 3.6.2
+              run_type: full
+              install_hole: True
+              codecov: True
+            - name: minimal-ubuntu
+              os: ubuntu-latest
+              python-version: 3.6
+              run_type: min
+              install_hole: False
+              codecov: True
+            - name: asv_check
+              os: ubuntu-latest
+              python-version: 3.6
+              run_type: full
+              install_hole: False
+              codecov: False
+    env:
+      CYTHON_TRACE_NOGIL: 1
+      MPLBACKEND: agg
+      MACOSX_DEPLOYMENT_TARGET: 10.9
+      GH_OS: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: setup_osx
+      if: startsWith(matrix.os, 'macOS')
+      run: |
+        echo "OS_NAME=osx" >> $GITHUB_ENV
+        ulimit -S -n 2048
+
+    - name: setup_linux
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        echo "OS_NAME=linux" >> $GITHUB_ENV
+
+    - name: setup_miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        auto-update-conda: true
+        channel-priority: flexible
+        # channel clash even though you have the same definition..
+        #channels: biobuilds, conda-forge
+        add-pip-as-python-dependency: true
+        # mamba does not currently support pip in env files
+        # to enable post gh/mamba-org/mamba/issues/589
+        #mamba-version: "*"
+        architecture: x64
+        environment-file: maintainer/test-envs/conda-${{ matrix.run_type }}.yaml
+        activate-environment: mda-tests
+    
+    - name: check_setup
+      run: |
+        # Check OS
+        echo "OS: ${OS_NAME}"
+        # Check python
+        which python
+        which pip
+        conda info
+        # TODO: adequate tqdm installed, setup.py still overrides...
+        conda list
+
+    - name: install_asv
+      if: contains(matrix.name, 'asv_check')
+      run: |
+        pip install asv "setuptools<45.0.0"
+
+    - name: install_hole
+      if : matrix.install_hole
+      run: |
+        # cert issue, anyone with a mac can circumvent this?
```

Comment:
fascinatingly, macOS stores its certificates in keychain access, which curl, being not being mac-specific, does not know. Adding this to the install_hole.sh exports the keychain certificates and tells curl where to find it:

```diff
OSNAME="$1"
case "$OSNAME" in
    Linux|linux)
	OSNAME=Linux
	OFORMAT=Linux
	DOWNLOAD_URL="${DOWNLOAD_URL_LINUX}"
	TARFILE=${TARFILE_LINUX}
+++    CURLARGS=""  # i think this dies if a variable is unset
	;;
    OSX|osx|Darwin|darwin)
	OSNAME=Darwin
	OFORMAT="Mach-O"
	DOWNLOAD_URL="${DOWNLOAD_URL_DARWIN}"
	TARFILE=${TARFILE_DARWIN}
+++    security export -p -t certs -k `security list-keychains -d system|cut -d '"' -f 2` -o certs.crt
+++    CURLARGS="--cacert $(PWD)/certs.crt"
	;;
    *)
	die "OS '${OSNAME}' not supported." 10;;
esac

PREFIX="$2"
test -n "$PREFIX" || die "missing second argument PREFIX (installation directory)" 10

#------------------------------------------------------------
# start installation
#------------------------------------------------------------

mkdir -p "$PREFIX" && cd "$PREFIX" || die "Failed to create and cd to $PREFIX" 1
if ! test -f ${TARFILE}; then
    echo "Downloading ${TARFILE} from ${DOWNLOAD_URL}..."
    # fixing curl on travis/anaconda, see http://stackoverflow.com/a/31060428/334357
    export CURL_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
---    curl -L ${DOWNLOAD_URL} -o ${TARFILE} $CURLARGS || \
+++    curl -L ${DOWNLOAD_URL} -o ${TARFILE} $CURLARGS || \
    die "Failed to download ${TARFILE} from ${DOWNLOAD_URL}" 1
fi
```

Reply:
Should we be doing this at the install_hole level or GH Actions, i.e. how likely are the Apple overlords going to decide to change things between osx versions?

Reply:
This is outdated now and instead downloads a different certificate file. I don't think that will change quickly, I hope, and ideally the URL will not get hijacked (I guess we could upload our own file for this?) If Apple changes anything like adding sensible default certificates, we could change it back to use the system ones.